lib/spack/spack/test/cmd/env.py

```diff
@@ -1030,6 +1030,35 @@ def noop(*args):
         assert x_read['z'] != y_read['z']
 
 
+def test_env_dir_stays_after_deactivation():  # TODO: do docs
```

Comment:
If you use the `working_env` fixture as an argument to the test, then you don't have to worry about your test polluting the user environment for other tests.

```suggestion
def test_env_dir_stays_after_deactivation(working_env):  # TODO: do docs
```